packages/arb-evm/evm/blockresult.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package evm
+
+import (
+	"github.com/offchainlabs/arbitrum/packages/arb-util/value"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+type OutputStatistics struct {
+	GasUsed      *big.Int
+	TxCount      *big.Int
+	EVMLogCount  *big.Int
+	AVMLogCount  *big.Int
+	AVMSendCount *big.Int
+}
+
+func (os *OutputStatistics) AsValue() value.Value {
+	// Static slice correct size, so error can be ignored
+	tup, _ := value.NewTupleFromSlice([]value.Value{
+		value.NewIntValue(os.GasUsed),
+		value.NewIntValue(os.TxCount),
+		value.NewIntValue(os.EVMLogCount),
+		value.NewIntValue(os.AVMLogCount),
+		value.NewIntValue(os.AVMSendCount),
+	})
+	return tup
+}
+
+type BlockInfo struct {
+	BlockNum      *big.Int
+	Timestamp     *big.Int
+	GasLimit      *big.Int
+	BlockStats    *OutputStatistics
+	ChainStats    *OutputStatistics
+	PreviousBlock *big.Int
+}
+
+func (b *BlockInfo) LastAVMLog() *big.Int {
+	return new(big.Int).Sub(b.ChainStats.AVMLogCount, big.NewInt(1))
+}
+
+func (b *BlockInfo) FirstAVMLog() *big.Int {
+	return new(big.Int).Sub(b.LastAVMLog(), b.BlockStats.AVMLogCount)
```

Comment:
It looks like you need to add 1 to this.  If the logs in this block are numbers 10, 11, and 12, then LastAVMLog() will be 12, AVMLogCount will be 3, so FirstAVMLog will return 9.  It should return 10.